source/loader/ze_loader.cpp

```diff
@@ -130,34 +130,29 @@ namespace loader
             debug_trace_message(message, "");
         }
         bool return_first_driver_result=false;
-        if(drivers.size()==1) {
+        std::string initName = "zeInit";
+        driver_vector_t *drivers = &zeDrivers;
+        // If this is sysmanOnly check_drivers, then zesInit is being called and we need to use zesDrivers.
+        if (sysmanOnly) {
+            drivers = &zesDrivers;
+            initName = "zesInit";
+        }
+        if(drivers->size()==1) {
             return_first_driver_result=true;
         }
 
-        for(auto it = drivers.begin(); it != drivers.end(); )
+        for(auto it = drivers->begin(); it != drivers->end(); )
         {
             std::string freeLibraryErrorValue;
             ze_result_t result = init_driver(*it, flags, globalInitStored, sysmanGlobalInitStored, sysmanOnly);
             if(result != ZE_RESULT_SUCCESS) {
-                if (it->handle) {
-                    auto free_result = FREE_DRIVER_LIBRARY(it->handle);
```

Comment:
zeInit or zesInit cannot unload the driver from the system because the driver library must be available for the other init.

ie zesInit may support 1 driver, while zeInit supports 2. Therefore, one cannot free the driver libraries until the end.